graphmap.go

```diff
@@ -15,15 +15,22 @@ type graphMap struct {
 	m sync.Map
 }
 
+// pipelineRegistration represents both linked nodes and the registration policy
+// for the pipeline.
+type pipelineRegistration struct {
```

Comment:
Naming is hard.  Perhaps, `registeredPipeline` since it's created as the result
of registering a pipeline.  As an added benefit, case-sensitive code searches
will find both `Pipeline` and `registeredPipeline`